Formula/lua-language-server.rb

```diff
@@ -0,0 +1,110 @@
+class LuaLanguageServer < Formula
+  desc "Language Server for the Lua language"
+  homepage "https://github.com/sumneko/lua-language-server"
+  # pull from git tag to get submodules
+  url "https://github.com/sumneko/lua-language-server.git",
+      tag:      "2.5.2",
+      revision: "cb2042160865589b5534a6bf0b6c366ae4ab1d99"
+  license "MIT"
+  head "https://github.com/sumneko/lua-language-server.git", branch: "master"
+
+  depends_on "ninja" => :build
+
+  on_linux do
+    depends_on "gcc" # For C++17
+  end
+
+  fails_with gcc: 5
+
+  # Disable `filesystem.test_appdata_path`.
+  #
+  # This test expeccts the application to run under the user's ${HOME},
+  # which is not compatible with homebrew's build environment.
+  # See https://github.com/actboy168/bee.lua/issues/21
+  # Adapted from https://github.com/macports/macports-ports/tree/master/lua/lua-language-server
+  patch :DATA
+
+  def install
+    ENV.cxx11
+    system "git", "submodule", "update", "--init", "--recursive"
+    chdir "3rd/luamake" do
+      system "compile/install.sh"
+    end
+    system "3rd/luamake/luamake", "rebuild"
+    bindir = if OS.mac?
+      "bin/macOS"
+    else
+      "bin/Linux"
```

Comment:
```suggestion
      "bin/#{OS.kernel_name}"
```

Reply:
is that guaranteed to be "Linux"? That is hard-coded in the build scripts...

Reply:
`OS.kernel_name` is just [a wrapper for `uname -s`](https://github.com/Homebrew/brew/blob/5096d6eab2f923796ff28c4ca68bd402edf2f1c7/Library/Homebrew/os.rb#L38-L44).

Reply:
In which case we should keep the literal `Linux`, since that is _always_ the directory used on any Linux system.

Reply:
Doesn't `uname -s` return Linux for any Linux system?

Reply:
I don't know ;) But what would be the benefit of relying on a function to return a string you know in advance?

Reply:
It should, because `uname -s` is supposed to return the name of the kernel. The benefit is that we can avoid hardcoding constants, which I think is always a good idea. It'll make `brew` easier to port to a different platform, for example. Not that that's happening anytime soon, but you never know!

Reply:
But lua-language-server _itself_ hardcodes these constants! I'd be different if we could use `uname` on _all_ platforms to avoid the conditional -- but we can't, since the constant for macOS is `macOS` and not `Darwin`.